clients/drcachesim/tools/record_filter_launcher.cpp

```diff
@@ -75,17 +119,43 @@ _tmain(int argc, const TCHAR *targv[])
                     droption_parser_t::usage_short(DROPTION_SCOPE_ALL).c_str());
     }
 
-    dynamorio::drmemtrace::record_filter_t::record_filter_func_t *null_filter =
-        new dynamorio::drmemtrace::null_filter_t();
-    std::vector<dynamorio::drmemtrace::record_filter_t::record_filter_func_t *>
+    std::vector<
+        std::unique_ptr<dynamorio::drmemtrace::record_filter_t::record_filter_func_t>>
         filter_funcs;
-    filter_funcs.push_back(null_filter);
+    if (op_cache_filter_size.specified()) {
+        filter_funcs.emplace_back(
+            std::unique_ptr<dynamorio::drmemtrace::record_filter_t::record_filter_func_t>(
+                // XXX: add more command-line options to allow the user to set these
+                // parameters.
+                new dynamorio::drmemtrace::cache_filter_t(
+                    /*cache_associativity=*/1, /*cache_line_size=*/64,
+                    op_cache_filter_size.get_value(),
+                    /*filter_data=*/true, /*filter_instrs=*/false)));
+    }
+    if (op_remove_trace_types.specified() || op_remove_marker_types.specified()) {
+        std::vector<trace_type_t> filter_trace_types =
+            parse_string<trace_type_t>(op_remove_trace_types.get_value());
+        std::vector<trace_marker_type_t> filter_marker_types =
+            parse_string<trace_marker_type_t>(op_remove_marker_types.get_value());
+        if (!filter_marker_types.empty())
+            filter_trace_types.push_back(TRACE_TYPE_MARKER);
```

Comment:
See above: putting this here seems to just cause confusion and complexity.

Reply:
Fixed.